server/src/internalClusterTest/java/org/elasticsearch/snapshots/RestoreSnapshotIT.java

```diff
@@ -158,6 +161,73 @@ public void testParallelRestoreOperationsFromSingleSnapshot() throws Exception {
         assertThat(client.prepareGet(restoredIndexName2, sameSourceIndex ? docId : docId2).get().isExists(), equalTo(true));
     }
 
+    @TestLogging(
+        reason = "testing the logging of the start and completion of a snapshot restore",
+        value = "org.elasticsearch.snapshots.RestoreService:INFO"
+    )
+    public void testRestoreLogging() throws IllegalAccessException {
+        final MockLogAppender mockLogAppender = new MockLogAppender();
+        try {
+            String indexName = "testindex";
+            String repoName = "test-restore-snapshot-repo";
+            String snapshotName = "test-restore-snapshot";
+            Path absolutePath = randomRepoPath().toAbsolutePath();
+            logger.info("Path [{}]", absolutePath);
+            String restoredIndexName = indexName + "-restored";
+            String expectedValue = "expected";
+
+            mockLogAppender.start();
+            Loggers.addAppender(LogManager.getLogger(RestoreService.class), mockLogAppender);
+
+            mockLogAppender.addExpectation(new MockLogAppender.LoggingExpectation() {
+                String startMessageSnapshot;
+                Object startMessageIndicesList;
+                String completionMessageSnapshot;
+
+                @Override
+                public void match(LogEvent event) {
+                    final String message = event.getMessage().getFormattedMessage();
+                    if (message.startsWith("started restore of snapshot")) {
+                        startMessageSnapshot = event.getMessage().getParameters()[0].toString();
+                        startMessageIndicesList = event.getMessage().getParameters()[1];
+                    } else if (message.startsWith("completed restore of snapshot")) {
+                        completionMessageSnapshot = event.getMessage().getParameters()[0].toString();
+                    }
+                }
+
+                @Override
+                public void assertMatched() {
+                    assertThat(startMessageSnapshot, startsWith(snapshotName));
+                    assertEquals(Arrays.asList(indexName), startMessageIndicesList);
+                    assertThat(completionMessageSnapshot, startsWith(snapshotName));
+                }
+            });
```

Comment:
I think it'd be clearer to write this as two regular `SeenEventExpectation` expectations with the actual message we expect (maybe with some `*` wildcards) instead of doing this. But maybe I've missed some subtlety about why we're doing it this way.

Done this way I think the fields would need to be `volatile` right?